### PR TITLE
Add new LiKafkaConsumerImpl metric - _offsetInvalidOrOutRangeCounter to estimate the potential data loss

### PR DESCRIPTION
**Pull request by zsong-li**
Added new metric( _offsetInvalidOrOutRangeCounter) into LiKafkaConsumerImpl to count number of OffsetOutOfRange exceptions under following cases with liclosest reset strategy.

**1. fetchedOffset <  Log Start Offset (LSO)
2. fetchedOffset <= Log End Offset (LEO)**

This metric can be used as indicator to estimate the potential data loss upon the LiKafkaConsumer client reseting the offset to "liclosest"

### CODE REVIEW

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java [hunk 1]
```diff
@@ -150,6 +154,29 @@ public Object metricValue() {
       }
     };
 
+    MetricName offsetOutOfRangeCounterName = new MetricName(
+        "consumer-offset-out-of-range-counter",
+        "lnkd",
+        "counter of how many times the consumer reached OffsetOutOfRange or Invalid Offset exceptions.",
```
Comment (navina) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java:160]: nit: It should be `OffsetOutOfRangeException` or `NoOffsetForPartitionException`.

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java [hunk 2]
```diff
@@ -150,6 +154,29 @@ public Object metricValue() {
       }
     };
 
+    MetricName offsetOutOfRangeCounterName = new MetricName(
+        "consumer-offset-out-of-range-counter",
```
Comment (navina) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java:158]: maybe called this `consumer-num-invalid-offset-exceptions`

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java [hunk 3]
```diff
@@ -689,10 +717,12 @@ private void handleLiClosestResetStrategy(InvalidOffsetException oe) {
 
       if (!seekBeginningPartitions.isEmpty()) {
         LOG.info("Offsets are out of range for partitions {}. Seeking to the beginning offsets returned", seekBeginningPartitions);
+        _offsetInvalidOrOutRangeCount.getAndIncrement();
```
Comment (navina) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java:720]: Why is the counter increment here as well?
I think it is worth clarifying what count we are referring to with this metrics - is it the number of times the underlying ("apache") client is throwing these exceptions? Or something else as well?

Comment (zsong-li) [reply li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java:720]: Yes, this metrics counts the total number of times the apache consumer clients throwing the exceptions of "OffsetOutOfRange", which marks the potential data loss

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java [hunk 4]
```diff
@@ -150,6 +154,30 @@ public Object metricValue() {
       }
     };
 
+    MetricName offsetOutOfRangeCounterName = new MetricName(
+        "consumer-num-invalid-offset-exceptions",
```
Comment (navina) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java:158]: sorry to nag: can we call this `consumer-liclosest-data-loss-estimate`

**Pull request merged**